### ISSUE TITLE
Create a separate option to exclude tests from the callers list when listing unused functions

### ISSUE THREAD

**Issue by raszi** [clojure-lsp/clojure-lsp#1878]
**Is your feature request related to a problem? Please describe.**
The problem is that if you have a good test coverage most of your functions will be covered by tests. This means that if you refactor the code and remove some of the functionality you don't need anymore, listing the unused functions would exclude them and therefore you miss removing functions only used by tests.

**Describe the solution you'd like**
I would like to see a warning about unused functions, even if they are only referenced from tests.

**Describe alternatives you've considered**
Using `:exclude-regex` or temporarily changing the `:source-paths` to only refer to `#{"src"}`.

**Additional context**
https://clojurians.slack.com/archives/CPABC1H61/p1725458933828109

**Comment by ericdallo** [comment]
@raszi what about a setting for that linter like `:ignore-tests true`?

**Comment by raszi** [comment]
Then it sounds good to me!

**Comment by raszi** [comment]
As I figured out the `:exclude-regex` is not an option, because that excludes warnings in the namespaces or specific vars.

**Comment by ericdallo** [comment]
This is done under the `ignore-test-references?` flag, available as a nightly build already

**Comment by ericdallo** [comment]
![image](https://github.com/user-attachments/assets/1f36ed8b-d409-45b9-93d0-01ebf95fccd8)

**Comment by raszi** [comment]
Awesome! Thank you very much for the quick turnaround!

I'll check this out tomorrow the latest.

**Comment by raszi** [comment]
It works like a charm! Thank you very much! 🎉

**Comment by ericdallo** [comment]
@raszi I had to[ revert this feature](https://github.com/clojure-lsp/clojure-lsp/commit/7aefddd1fbb2664b27614f1a6081f75c554213d3) because of a huge decrease in performance, we will work on this fix soon
c/c @acamargo

**Comment by ericdallo** [comment]
Yeah, I agree, the problem is that that piece of code is pretty critical to the whole diagnostics performance and it's hard not to affect performance drastically, but we will think about something

**Comment by raszi** [comment]
I guess then this issue should be reopened, not?

**Comment by ericdallo** [comment]
@acamargo I believe we fixed again this, right?
